build_runner_core/lib/src/generate/input_matcher.dart

```diff
@@ -32,27 +32,27 @@ class InputMatcher {
   /// If there are no [includeGlobs] this always returns `true`.
   bool includes(AssetId input) =>
       includeGlobs == null ||
-      includeGlobs.isEmpty ||
-      includeGlobs.any((g) => g.matches(input.path));
+      includeGlobs!.isEmpty ||
+      includeGlobs!.any((g) => g.matches(input.path));
```

Comment:
I think it would be worth a local variable here and `excludes`
 
 ```dart
 bool includes(AssetId input) {
   final includeGlobs = this.includeGlobs;
   return includeGlobs == null ||
     includeGlobs.isEmpty ||
     includeGlobs.any((g) => g.matches(input.path));
 }

Reply:
done